kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/CSVUtil.java

```diff
@@ -71,36 +77,45 @@ public static CSVWriter newWriter(OutputStream outgoing, CSVProperties props) {
   private static final Pattern FLOAT = Pattern.compile("\\d*\\.\\d*[fF]?");
   private static final int DEFAULT_INFER_LINES = 25;
   private static final Set<String> NO_REQUIRED_FIELDS = ImmutableSet.of();
+  private static final HashMap<String,String> NO_SPECIFIED_TYPES = null;
 
   public static Schema inferNullableSchema(String name, InputStream incoming,
                                            CSVProperties props)
       throws IOException {
-    return inferSchemaInternal(name, incoming, props, NO_REQUIRED_FIELDS, true);
+    return inferSchemaInternal(name, incoming, props, NO_REQUIRED_FIELDS, NO_SPECIFIED_TYPES, true);
   }
 
   public static Schema inferNullableSchema(String name, InputStream incoming,
                                            CSVProperties props,
                                            Set<String> requiredFields)
       throws IOException {
-    return inferSchemaInternal(name, incoming, props, requiredFields, true);
+    return inferSchemaInternal(name, incoming, props, requiredFields, NO_SPECIFIED_TYPES, true);
+  }
+
+  public static Schema inferNullableSchema(String name, InputStream incoming,
```

Comment:
Yes, we are probably going to make this a public module at some point so it would be great to have the method. Thanks!